pkg/util/unstructured.go

```diff
@@ -77,28 +77,20 @@ func matchesGVK(o interface{}, group, version, kind string) bool {
 	}
 
 	if gvk.Group == group &&
-		gvk.Version == version &&
+		(version == AnyVersion || gvk.Version == version) &&
 		gvk.Kind == kind {
 		return true
 	}
 	return false
 }
 
 func getObjectGVK(o interface{}) (schema.GroupVersionKind, error) {
-	gvk := schema.GroupVersionKind{}
 	switch obj := o.(type) {
 	case runtime.Object:
-		gvk.Group = obj.GetObjectKind().GroupVersionKind().Group
-		gvk.Version = obj.GetObjectKind().GroupVersionKind().Version
-		gvk.Kind = obj.GetObjectKind().GroupVersionKind().Kind
-	case unstructured.Unstructured:
-		gvk.Group = obj.GetObjectKind().GroupVersionKind().Group
-		gvk.Version = obj.GetObjectKind().GroupVersionKind().Version
-		gvk.Kind = obj.GetObjectKind().GroupVersionKind().Kind
-	default:
-		return gvk, fmt.Errorf("object is of wrong type: %#v", obj)
-	}
-	return gvk, nil
+		return obj.GetObjectKind().GroupVersionKind(), nil
```

Comment:
Removing the case for `unstructured.Unstructured` may have unintended consequences on other parts of the code using this function. I had scenarios where the object could only be cast to `unstructured.Unstructured` but not to `runtime.Object`.

Reply:
how is that possible?
`Unstructured` implements `runtime.Object` interface.
so if it can be cast to `unstructured` is should be possible to cast also to `runtime.Object` interface.

Reply:
You are correct. I remember having that issue once but now I cannot reproduce. Maybe it was about creating directly an object as `unstructured.Unstructured` which did not have the deep copy functions and therefore the cast threw an error saying it was missing those. But I did a quick test with this change and I don't see that happing, so I will go ahed and approve.

Reply:
`func (in *Unstructured) DeepCopy() *Unstructured` exists.

Reply:
So does `func (obj *Unstructured) GetObjectKind() schema.ObjectKind`.  Those are the two methods that make up `runtime.Object`.

Reply:
The following code compiles.

```
	test := &unstructured.Unstructured{}
	var testR runtime.Object = test
```